theano/gpuarray/opt.py

```diff
@@ -2069,7 +2085,7 @@ def provide_inputs(self, node, inputs):
                                          node.op.subsample,
                                          node.op.filter_dilation)
 
-            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[2:])
+            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[-2:])
```

Comment:
but I think it would be a good idea to test the default optimisers as well using the meta-optimizer to prevent it from happening again.
@nouiz @gvtulder what do you think?